src/peripheral/plic.rs

```diff
@@ -0,0 +1,230 @@
+//! Platform-Level Interrupt Controller (PLIC) peripheral.
+//!
+//! Specification: https://github.com/riscv/riscv-plic-spec/blob/master/riscv-plic.adoc
+
+use super::PLIC;
+use volatile_register::{RO, RW};
+
+/// Maximum number of interrupt sources supported by the PLIC standard.
+const MAX_SOURCES: usize = 1_024;
+/// Maximum number of words needed to represent interrupts with flags.
+const MAX_FLAGS_WORDS: usize = MAX_SOURCES / (u32::BITS as usize);
+/// Maximum number of contexts supported by the PLIC standard.
+const MAX_CONTEXTS: usize = 15_872;
+
+/// Register block.
+#[repr(C)]
+pub struct RegisterBlock {
+    /// `0x0000_0000..=0x0000_0FFC` - Interrupt Priority Register.
+    pub priority: [RW<u32>; MAX_SOURCES],
```

Comment:
Priority 0x0 is reserved, maybe a check should be added to prevent it being used?

Reply:
According to the standard, "a priority value of 0 is reserved to mean "never interrupt" and effectively disables the interrupt". So I think that 0 is a valid priority that disables the interruption indirectly.

Reply:
I see, yep, I think you are right on this.